### ISSUE TITLE
Error on iter_groups() method -> " ValueError: arrays must all be same length" 

### ISSUE THREAD

**Issue by AkbarAlam** [danielhrisca/asammdf#97]
Hi 

I am getting an error if I apply the following code 

```
for df in mdf.iter_groups():
    print(df)
```

the error : 

```
---------------------------------------------------
ValueError        Traceback (most recent call last)
<ipython-input-88-9207fd0496f7> in <module>()
----> 1 for df in mdf.iter_groups():
      2     print(df)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\asammdf\mdf.py in iter_groups(self)
   2358                 pandas_dict[name] = np.concatenate(sig)
   2359 
-> 2360             yield DataFrame.from_dict(pandas_dict)
   2361 
   2362     def resample(self, raster, memory=None, version=None):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py in from_dict(cls, data, orient, dtype, columns)
    983             raise ValueError('only recognize index or columns for orient')
    984 
--> 985         return cls(data, index=index, columns=columns, dtype=dtype)
    986 
    987     def to_dict(self, orient='dict', into=dict):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py in __init__(self, data, index, columns, dtype, copy)
    346                                  dtype=dtype, copy=copy)
    347         elif isinstance(data, dict):
--> 348             mgr = self._init_dict(data, index, columns, dtype=dtype)
    349         elif isinstance(data, ma.MaskedArray):
    350             import numpy.ma.mrecords as mrecords

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py in _init_dict(self, data, index, columns, dtype)
    457             arrays = [data[k] for k in keys]
    458 
--> 459         return _arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)
    460 
    461     def _init_ndarray(self, values, index, columns, dtype=None, copy=False):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py in _arrays_to_mgr(arrays, arr_names, index, columns, dtype)
   7313     # figure out the index, if necessary
   7314     if index is None:
-> 7315         index = extract_index(arrays)
   7316 
   7317     # don't force copy because getting jammed in an ndarray anyway

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py in extract_index(data)
   7359             lengths = list(set(raw_lengths))
   7360             if len(lengths) > 1:
-> 7361                 raise ValueError('arrays must all be same length')
   7362 
   7363             if have_dicts:

ValueError: arrays must all be same length
```

That worked before the new release with all sort of  MDF files as far I can remember. 
Am I doing something wrong? 

It is pretty easier to work with Pandas data frame that's why I need to use this method.

**Comment by danielhrisca** [comment]
Hello Akbar,

 please check the latest development commit. The inconvenient is that invalidation bits are not considered.

**Comment by danielhrisca** [comment]
Hello Akbar,

is it working correctly now?

**Comment by AkbarAlam** [comment]
Oh I haven't tried, I was wondering if you are going to add it in the next release, however, i have figured out another way to solve the thing i was doing

**Comment by danielhrisca** [comment]
the fix is included in release 4.1.2
